src/app/shared/ag-grid-renderers/rating-renderer/rating-renderer.component.ts

```diff
@@ -0,0 +1,43 @@
+import { Component } from '@angular/core';
+import { AgRendererComponent } from 'ag-grid-angular';
+import { RATING_STATES, RATING_STATES_LABEL } from '../../../../constants/app-constants';
+import { ICellRendererParams } from 'ag-grid';
+
+@Component({
+  selector: 'app-rating-renderer',
+  templateUrl: './rating-renderer.component.html',
+  styleUrls: ['./rating-renderer.component.scss']
+})
+export class RatingRendererComponent implements AgRendererComponent {
+    private params: ICellRendererParams;
+    value: any;
+    styles: any;
+
+    agInit(params: any): void {
+        this.params = params;
+        this.value = RATING_STATES_LABEL[this.params.value];
+
+        this.styles = {
+            width: this.value + '%',
```

Comment:
this.value is a text string, why are we using a text string for the width value?